### PR TITLE
threading mg execution and ensuring that we alway exit 0 (true)

### PR DESCRIPTION
**Pull request by sferich888**
@RickJWagner @coreydaley PTAL
@sanchezl  you may want to review as well.

### CODE REVIEW

**Comment by RickJWagner** [comment]
In 'gather', do we really want to make the same directory in a loop?  It seems we'd make it once and have it.  

for resource in ${resources[@]}; do
    LOG_DIR="${BASE_COLLECTION_PATH}/must-gather-resource-logs"
    mkdir -p ${LOG_DIR}

File: collection-scripts/gather [hunk 1]
```diff
@@ -24,13 +25,22 @@ resources+=(clusternetworks hostsubnets)
 # Autoscaler Resources
 resources+=(clusterautoscaler machineautoscaler)
 
+PIDS=()
 # Run the Collection of Resources using must-gather
 for resource in ${resources[@]}; do
-    /usr/bin/openshift-must-gather inspect ${resource}
+    LOG_DIR="${BASE_COLLECTION_PATH}/must-gather-resource-logs"
+    mkdir -p ${LOG_DIR}
+    /usr/bin/openshift-must-gather inspect ${resource} 2> ${LOG_DIR}/${resource/\//_}.log &
```
Comment (sanchezl) [inline collection-scripts/gather]:
```suggestion
    /usr/bin/openshift-must-gather inspect ${resource} 2> ${LOG_DIR}/${resource//[^A-Za-z0-9._-]/_}.log &
```

Comment (sferich888) [reply collection-scripts/gather]: I don't understand the sugestion, my code converts resources with '/' in them like `ns/default` to resources names with '_' in them like `ns_default`, what is yours doing?

This avoid bash path issues when I go to save the resource.

Comment (sferich888) [reply collection-scripts/gather:33]: @sanchezl can you followup here?

**Comment by sferich888** [comment]
@RickJWagner I fixed the log directory creation, happening in a loop.

**Comment by sferich888** [comment]
Obsolete

**Pull request closed without merging**